cli/batch/forms.py

```diff
@@ -0,0 +1,87 @@
+"""
+Provide forms for command line interface's batch commands.
+"""
+import re
+
+from marshmallow import (
+    Schema,
+    ValidationError,
+    fields,
+    validates,
+)
+
+from cli.constants import (
+    DOMAIN_NAME_REGEXP,
+    HEADER_SIGNATURE_REGEXP,
+)
+
+
+class ListBatchesForm(Schema):
+    """
+    List batches validation form.
+    """
+
+    ids = fields.List(fields.String(), allow_none=True, required=False)
+    start = fields.String(allow_none=True, required=False)
+    limit = fields.Integer(allow_none=True, required=False)
+    head = fields.String(allow_none=True, required=False)
+    reverse = fields.String(allow_none=True, required=False)
+    node_url = fields.String(allow_none=True, required=False)
+
+    @validates('ids')
+    def validate_ids(self, ids):
+        """
+        Validate batch identifiers.
+        """
+        if ids is not None:
+            for batch_id in ids:
+                if re.match(pattern=HEADER_SIGNATURE_REGEXP, string=batch_id) is None:
+                    raise ValidationError(f'The following batch id `{batch_id}` is invalid.')
+
+    @validates('start')
+    def validate_start(self, start):
+        """
+        Validate start.
+        """
+        if start is not None and re.match(pattern=HEADER_SIGNATURE_REGEXP, string=start) is None:
+            raise ValidationError(f'The following batch id `{start}` is invalid.')
+
+    @validates('limit')
+    def validate_limit(self, limit):
+        """
+        Validate limit.
+        """
+        if limit is not None and limit < 0:
+            raise ValidationError("Limit can't be negative.")
+
+    @validates('head')
+    def validate_head(self, head):
+        """
+        Validate head.
+        """
+        if head is not None and re.match(pattern=HEADER_SIGNATURE_REGEXP, string=head) is None:
+            raise ValidationError(f'The following block id `{head}` is invalid.')
+
+    @validates('reverse')
+    def validate_reverse(self, reverse):
```

Comment:
Please, create [custom fields](https://github.com/Remmeauth/remme-core-cli/blob/develop/cli/generic/forms/fields.py) instead of a validation function for now. 

We already did the migrations to them.